src/core/parse_worker.hpp

```diff
@@ -0,0 +1,132 @@
+#pragma once
+
+#include <string>
+#include <unordered_set>
+#include <vector>
+#include <nlohmann/json.hpp>
+#include "symbol.hpp"
+#include "token.hpp"
+
+namespace docgen {
+namespace core {
+
+class ParseWorker;
+
+using symbol_t = Symbol;
+using token_t = Token<symbol_t>;
+using routine_t = void (*)(const token_t& t);
+using worker_t = ParseWorker;
+
+/*
+ * A ParseWorker object holds an ordered array of SymbolHandler objects;
```

Comment:
Order is important because each one is handled in order—for example, in a `CommentParser`, there might be a `SymbolHandler` for `/*` and one for `*/`. The order in which they positioned in the array matters a lot—if you want the `CommentParser` to look for `/*` first, then it has to be first in the array. It should first handle `/*`, and then `*/`. Hence, "ordered", as in "order matters".

Reply:
Ohh got it I thought you were like sorting them or something like comparing symbol handlers.